### PR TITLE
Development auth

### PR DESCRIPTION
**Pull request by dmilburn**
@kennygeiler @danasselin 

This commit includes basic auth functionality (login, logout, signup). I made login and signup separate pages but that can be easily changed if you guys disagree. There's a flash error message in the layout.erb file. FYI the links on the index page are currently set to #.

### CODE REVIEW

File: source/app/controllers/auth.rb [hunk 1]
```diff
@@ -1,11 +1,49 @@
...
+  if current_user
+    redirect '/'
+  else
+    erb :'auth/signup'
+  end
+end
 
+post '/signup' do
+  user = User.new(params[:user])
+  if empty_sign_up_field?(params[:user])
```
Comment (sgharms) [inline source/app/controllers/auth.rb:32]: extract to a helper?

``` ruby
def some_helper(params)
  flash[:error] = #logic
end
```

File: source/app/helpers/auth.rb [hunk 2]
```diff
@@ -5,3 +5,10 @@ def current_user
     return nil
   end
 end
+
+def empty_sign_up_field?(user_params)
+  user_params.each_value do |param|
```
Comment (sgharms) [inline source/app/helpers/auth.rb:10]: There are Enumerables better than this.

**Comment by sgharms** [comment]
There are some changes that I'd recommend to this....

File: source/app/views/auth/signup.erb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<form action="/signup" method="POST">
+  <label>Username:</label>
+  <input type="text" name="user[name]">
```
Comment (dmilburn) [inline source/app/views/auth/signup.erb:3]: value="<%= @user.name %>"

**Pull request merged**